src/components/LeftPane/Drawers/AnalyticsDrawer/ServiceAreaAnalytics.tsx

```diff
@@ -1,11 +1,41 @@
 import * as React from 'react'
 import { withStore } from '../../../../services/store'
+import { averageDistance, averageTime, maxDistance, maxTime, minDistance, minTime } from '../../../../utils/analytics'
+import { adequaciesFromServiceArea } from '../../../../utils/data'
+import { formatNumber } from '../../../../utils/formatters'
 import { StatsBox } from '../../../StatsBox/StatsBox'
 import { AdequacyDoughnut } from './AdequacyDoughnut'
 import './ServiceAreaAnalytics.css'
 
-export let ServiceAreaAnalytics = withStore('selectedServiceArea')(({ store }) =>
-  <div className='ServiceAreaAnalytics'>
+export let ServiceAreaAnalytics = withStore(
+  'adequacies',
+  'selectedServiceArea'
+)(({ store }) => {
+
+  let selectedServiceArea = store.get('selectedServiceArea')!
+  let adequacies = adequaciesFromServiceArea(selectedServiceArea, store)
+
+  return <div className='ServiceAreaAnalytics'>
     <AdequacyDoughnut serviceAreas={[store.get('selectedServiceArea')!]} />
+    <StatsBox withBorders withHorizontalLines>
+      <tr>
+        <th />
+        <th>Avg</th>
+        <th>Min</th>
+        <th>Max</th>
+      </tr>
+      <tr>
+        <th>Distance (mi)</th>
+        <td>{formatNumber(averageDistance(adequacies))}</td>
+        <td>{formatNumber(minDistance(adequacies))}</td>
+        <td>{formatNumber(maxDistance(adequacies))}</td>
+      </tr>
+      <tr>
+        <th>Time (min)</th>
+        <td>{formatNumber(averageTime(adequacies))}</td>
```

Comment:
For the future, but we could probably have a function that loops through the data once only and outputs these 3 metrics

Reply:
Agreed! Today I use a transducer to knock the *O(2n)* computation down to *O(n)*, but you're right that we repeat it 5-6 times anyway.

It will take some thought for how to do it in *O(1)* time cleanly.